torch/distributed/checkpoint/planner.py

```diff
@@ -426,3 +427,31 @@ def commit_tensor(self, read_item: ReadItem, tensor: torch.Tensor) -> None:
         The contents of tensor will follow its device synchronization model.
         """
         pass
+
+
+class CheckpointableTensor(abc.ABC):
```

Comment:
I've been leaning against using abstract classes these days. It makes it harder to maintain both consistency and backwards compatibility when we add new methods, but maybe that's just me.

Does anyone else have this opinion?

Reply:
I agree, latest revision removes this. And there was actually some compatibility issues with abc.ABC and torch.Tensor subclasses, which removing helps resolve as well.